src/javascripts/Services/gridService.js

```diff
@@ -0,0 +1,90 @@
+// Base view to show nodes and columns. It's like a tree view but with no depth. 
+// Utilized by all views. Functions for specialized views (e.g. treeView/recordView)
+// will overwrite most of these base functions.
+ngapp.service('gridService', function($timeout, htmlHelpers) {
+    this.buildFunctions = function(scope, element) {
+        // scope functions
+        scope.reload = function() {
+            if (!scope.tree) return;
+            let oldTree = scope.tree;
+            scope.clearSelection(true);
+            scope.buildTree();
+            scope.releaseTree(oldTree);
+        };
+
+        scope.resolveNodeError = (path, part) => {
+            return new Error(`Failed to resolve node "${part}" in path "${path}"`);
+        };
+
+        scope.onNodeMouseDown = function(e, node) {
+            if (e.button !== 2 || !node.selected) scope.selectNode(e, node);
+            if (e.button === 2) scope.showContextMenu(e);
+        };
+
+        scope.resolveElements = function() {
```

Comment:
is this function actually used?  I don't think we need these elements for anything, currently?